### PR TITLE
ApproximateEqual

### LINKED ISSUE #11564: ApproximateEqual

### PR DESCRIPTION
**Pull request by shreyjain26**
Close #11564

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 1]
```diff
@@ -66,6 +66,33 @@ def test_tensorflow_Acosh(  # NOQA
...
+    on_device,
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:92]: please also add the arguments `y` and `tolerance`, hints on data generation will be added as another comment

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 2]
```diff
@@ -66,6 +66,33 @@ def test_tensorflow_Acosh(  # NOQA
     )
 
 
+# ApproximateEqual
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.ApproximateEqual",
+    dtype_and_x=helpers.dtype_and_values(
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:72]: to generate 2 arrays, add the parameter: `num_arrays=2`

as the types of the arrays for this function have to be the same, please add this parameter too: `shared_dtype=True`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 3]
```diff
@@ -66,6 +66,33 @@ def test_tensorflow_Acosh(  # NOQA
...
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
+    )
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:93]: then you'll be able to do `y=x[1]` here

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 4]
```diff
@@ -66,6 +66,33 @@ def test_tensorflow_Acosh(  # NOQA
     )
 
 
+# ApproximateEqual
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.ApproximateEqual",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+    ),
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:74]: data for tolerance can be generated such as: `tol=st.floats(1e-05, 1e-03)`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 5]
```diff
@@ -66,6 +66,33 @@ def test_tensorflow_Acosh(  # NOQA
...
+    test_with_out=st.just(False),
+)
+def test_tensorflow_ApproximateEqual(  # NOQA
+    *,
+    dtype_and_x,
+    frontend,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:80]: please remember to add `tol` in the list after generating the data

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py (hunk 3 above)
Comment (juliagsy) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:93]: then you can do `tolerance=tol` here after generating the data

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Great work so far! However, there are some required arguments missing in the tests, I've added hints as comments. Once you're done with the changes, please request my review again, thanks!

**Comment by shreyjain26** [comment]
I made the changes but the lint test is still failing.

File: ivy/functional/frontends/tensorflow/raw_ops.py [hunk 6]
```diff
@@ -43,6 +43,19 @@ def AddV2(*, x, y, name="AddV2"):
...
+@to_ivy_arrays_and_back
+def ApproximateEqual(
+    *,
+    x,
+    y,
+    tolerance=1e-05,
+    name="ApproximateEqual",
+):
+    x, y = check_tensorflow_casting(x, y)
+    ret = ivy.abs(x-y)
```
Comment (juliagsy) [inline ivy/functional/frontends/tensorflow/raw_ops.py:55]: the lint failures may be caused by this, it needs spaces in between like `x - y`, but this is not a major issue as I think we have a workflow for fixing this, feel free to push a fix though :)

File: ivy/functional/frontends/tensorflow/raw_ops.py [hunk 7]
```diff
@@ -43,6 +43,19 @@ def AddV2(*, x, y, name="AddV2"):
     return ivy.addv2(tf_frontend.to_tensor(x), tf_frontend.to_tensor(y))
 
 
+@to_ivy_arrays_and_back
+def ApproximateEqual(
```
Comment (juliagsy) [inline ivy/functional/frontends/tensorflow/raw_ops.py:47]: there is a list of limited accepted `dtypes` for this function, please check the docs for more info: https://www.tensorflow.org/api_docs/python/tf/raw_ops/ApproximateEqual

then add the `@with_unsupported_dtypes` accordingly

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Great work so far! The tests are not passing unfortunately due to a few more additions required! Btw, there is an import error in the workflow and tests are not being run in the workflow, so please run the tests locally and make sure that they are passing!

PS: I was meant to say that because of an issue in our workflow, though there are green ticks in the intelligent tests, there actually isn't any tests being run, so the ticks are not accurate as of now! Please do run them locally and verify that they are passing correctly! Thanks a lot!

**Comment by shreyjain26** [comment]
@juliagsy I'm getting two fails after putting in `with_unsupported_dtypes`

File: ivy/functional/frontends/tensorflow/raw_ops.py [hunk 8]
```diff
@@ -43,6 +43,34 @@ def AddV2(*, x, y, name="AddV2"):
     return ivy.addv2(tf_frontend.to_tensor(x), tf_frontend.to_tensor(y))
 
 
+@with_unsupported_dtypes(
+    {
+        "2.10.0 and below": (
+            "float16"
+            "bool",
```
Comment (juliagsy) [inline ivy/functional/frontends/tensorflow/raw_ops.py:50]: only `float16` and `bool` is necessary here, the others may be removed. please also add `bfloat16`, the tests should be passing after that, but please verify it again locally!

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Thanks for the great work! I've left some suggestions in the comments! As usual, please request my review again when you're done! Thanks a lot!

**Comment by shreyjain26** [comment]
I ran the test locally and I got this error 
image
Also, I realised that my environment is different from ivy's as mine is python 3.10 while ivy is using 3.8.10 I think that may alter my results.
image
Locally, the lint check/flask8 is passing but here it isn't.

**Review by juliagsy — COMMENTED** [review]
Hey there! Thanks for the changes! For the lint checks, it may be other parts of the repo failing and not related to your changes, so don't worry about that!

Not sure if it is related, but please remove any changes related to the `.idea` folder, please only keep the changes related to the function you are working on in this PR!

EDIT: I've also tried out the tests and they're working good for me! Once the `.idea` related changes are removed, we'll be able to merge this PR in general! As usual, please request my review again once the changes are done! Thanks so much!

**Comment by shreyjain26** [comment]
@juliagsy There was a change in .idea when I pushed from pycharm directly. But I had already corrected it 11 hours ago. The .idea files in my pr and unifyai: master are identical.

**Comment by shreyjain26** [comment]
In my other pull requests where I haven't made any changes in .idea file, even in those only lint is failing rest all tests are passing.

**Review by juliagsy — APPROVED** [review]
oh! it was only an end of file difference, apologies for overlooking it, my bad! thanks for contributing!

**Comment by shreyjain26** [comment]
Thanks a lot for the help 😄

**Pull request merged**